java/clients/src/test/java/sleeper/clients/admin/InstancePropertyReportTest.java

```diff
@@ -70,29 +71,56 @@ public void shouldPrintInstancePropertyReportWhenChosen() {
                         "# is 11. It can be overridden on a per-table basis.\n" +
                         "# (NB This does not apply to splitting jobs which will run even if there is only 1 file.)\n" +
                         "# This is a default value and will be used if not specified in the table.properties file\n" +
-                        "sleeper.default.compaction.files.batch.size: 11");
+                        "sleeper.default.compaction.files.batch.size: 11")
+                // Then check property groups are printed
+                .contains(PropertyGroup.all().stream()
+                        .map(PropertyGroup::getDescription)
+                        .toArray(String[]::new));
 
 
-        // Then check the ordering of some property names are correct
+        // Then check the order of some property names are correct
         assertThat(output.indexOf("sleeper.account"))
                 .isLessThan(output.indexOf("sleeper.log.retention.days"))
                 .isLessThan(output.indexOf("sleeper.vpc"));
-        assertThat(output.indexOf("sleeper.log.retention.days"))
-                .isLessThan(output.indexOf("sleeper.vpc"));
+        assertThat(output.indexOf("sleeper.ingest"))
+                .isLessThan(output.indexOf("sleeper.compaction"));
+        // Then check the order of some groups are correct
+        assertThat(output.indexOf(PropertyGroup.INGEST.getDescription()))
+                .isLessThan(output.indexOf(PropertyGroup.BULK_IMPORT.getDescription()));
+        assertThat(output.indexOf(PropertyGroup.GARBAGE_COLLECTOR.getDescription()))
+                .isLessThan(output.indexOf(PropertyGroup.COMPACTION.getDescription()));
+        // Then check that some UserDefinedProperties are in the correct group
+        assertThat(output.indexOf("sleeper.ingest.task.cpu"))
+                .isBetween(
+                        output.indexOf(PropertyGroup.INGEST.getDescription()),
+                        output.indexOf(PropertyGroup.BULK_IMPORT.getDescription()));
+        assertThat(output.indexOf("sleeper.compaction.job.creation.memory"))
+                .isBetween(
+                        output.indexOf(PropertyGroup.COMPACTION.getDescription()),
+                        output.indexOf(PropertyGroup.QUERY.getDescription()));
+        // Then check that SystemDefinedProperty is in the correct group
+        assertThat(output.indexOf("sleeper.query.role.arn"))
+                .isBetween(
+                        output.indexOf(PropertyGroup.QUERY.getDescription()),
+                        output.indexOf(PropertyGroup.LOGGING.getDescription()));
+        assertThat(output.indexOf("sleeper.partition.splitting.queue.url"))
+                .isBetween(
+                        output.indexOf(PropertyGroup.PARTITION_SPLITTING.getDescription()),
+                        output.indexOf(PropertyGroup.GARBAGE_COLLECTOR.getDescription()));
```

Comment:
We could split up this test and separate out the assertions.

It's not clear from these assertions how likely it is we'd get a false positive. If the description is quite short or reused it might not be a valid assertion. It's probably better to use an actual value in the test.

It looks like we'd get the same value from the tests if we only test one property is in the correct group for each type of property.